ckan/lib/helpers.py

```diff
@@ -685,12 +685,25 @@ def are_there_flash_messages():
 
 def _link_active(kwargs):
     ''' creates classes for the link_to calls '''
+    if is_flask_request():
+        return _link_active_flask(kwargs)
+    else:
+        return _link_active_pylons(kwargs)
+
+
+def _link_active_pylons(kwargs):
     highlight_actions = kwargs.get('highlight_actions',
                                    kwargs.get('action', '')).split()
     return (c.controller == kwargs.get('controller')
             and c.action in highlight_actions)
 
 
+def _link_active_flask(kwargs):
```

Comment:
We should avoid using Pylons terms on Flask functions. `controller` and `action` are Pylons terms that not apply to Flask, so in the context of a Flask function let's use:

```python
blueprint, endpoint = request.url_rule.endpoint.split('.')
```

---

Of course we need to check against kwargs.get('controller') and kwargs.get('action') to keep backwards compatibility, but we also need to introduce the new Flask-based equivalent to be ready to switch calls to `nav_link()` to the Flask syntax, eg something like:

```python
if kwargs.get('endpoint') == request.url_rule.endpoint:
```
So we can go from this

```
{{ h.nav_link(_('Login'), controller='user', action='login') }}
```

to this


```
{{ h.nav_link(_('Login'), endpoint='user.login') }}
```

I suggest we do this in one go when we update all `url_for`, `nav_link` etc calls after all controllers have been migrated, to avoid having function calls in the two syntaxes.